org-node/src/client.rs

```diff
@@ -282,69 +513,33 @@ impl Client {
                 let mut peers = api.providers(urn.clone(), timeout);
 
                 while let Some(peer) = peers.next().await {
-                    match Client::track_project(api, &urn, &peer).await {
-                        Err(err) => {
-                            tracing::error!(target: "org-node", "Error tracking {}: {}", urn, err);
+                    // Send notification to track peers process;
+                    let result = Client::track_peer_handler(
+                        peer_tx.clone(),
+                        TrackPeer {
+                            api: api.clone(),
+                            urn: urn.clone(),
+                            peer_info: peer,
+                            paths: paths.clone(),
+                        },
+                    )
+                    .await;
+
+                    // Handle the track peer response
+                    let response = match result {
+                        Ok(response) => response,
+                        Err(e) => {
+                            // QUESTION: Should we reply with an error here or continue?;
+                            tracing::error!(target: "org-node", "Error tracking peers: {:?}", e);
+                            continue;
```

Comment:
Should we be returning an error here instead of continuing?

```
reply
.send(Err(TrackProjectError::<REASON>))
.map_err(|_| Error::Reply("TrackProject".to_string()))
```

Reply:
No, we return a `NotFound` error if none of the peers we try succeed. It's okay if some of the attempts fail, we just care if one succeeds.

Reply:
eaa3a24 removes question comment.